modules/aolBidAdapter.js

```diff
@@ -1,341 +1,346 @@
-    }
-    return nexageApi;
-  }
+function _parseBid(response, bid) {
+  let bidData;
 
-  function _addErrorBidResponse(bid, response = {}) {
-    const bidResponse = bidfactory.createBid(2, bid);
-    bidResponse.bidderCode = bid.bidder;
-    bidResponse.reason = response.nbr;
-    bidResponse.raw = response;
-    bidmanager.addBidResponse(bid.placementCode, bidResponse);
+  try {
+    bidData = response.seatbid[0].bid[0];
+  } catch (e) {
+    return;
   }
 
-  function _addBidResponse(bid, response) {
-    let bidData;
+  let cpm;
 
-    try {
-      bidData = response.seatbid[0].bid[0];
-    } catch (e) {
-      _addErrorBidResponse(bid, response);
-      return;
-    }
-
-    let cpm;
+  if (bidData.ext && bidData.ext.encp) {
+    cpm = bidData.ext.encp;
+  } else {
+    cpm = bidData.price;
 
-    if (bidData.ext && bidData.ext.encp) {
-      cpm = bidData.ext.encp;
-    } else {
-      cpm = bidData.price;
-
-      if (cpm === null || isNaN(cpm)) {
-        utils.logError('Invalid price in bid response', AOL_BIDDERS_CODES.aol, bid);
-        _addErrorBidResponse(bid, response);
-        return;
-      }
+    if (cpm === null || isNaN(cpm)) {
+      utils.logError('Invalid price in bid response', AOL_BIDDERS_CODES.AOL, bid);
+      return;
     }
+  }
 
-    let ad = bidData.adm;
-    if (response.ext && response.ext.pixels) {
-      if (bid.params.userSyncOn === constants.EVENTS.BID_RESPONSE) {
-        dropSyncCookies(response.ext.pixels);
-      } else {
-        let formattedPixels = response.ext.pixels.replace(/<\/?script( type=('|")text\/javascript('|")|)?>/g, '');
-
-        ad += '<script>if(!parent.$$PREBID_GLOBAL$$.aolGlobals.pixelsDropped){' +
-          'parent.$$PREBID_GLOBAL$$.aolGlobals.pixelsDropped=true;' + formattedPixels +
-          '}</script>';
-      }
-    }
+  let ad = bidData.adm;
+  if (response.ext && response.ext.pixels) {
+    if (config.getConfig('aol.userSyncOn') !== constants.EVENTS.BID_RESPONSE) {
+      let formattedPixels = response.ext.pixels.replace(/<\/?script( type=('|")text\/javascript('|")|)?>/g, '');
 
-    const bidResponse = bidfactory.createBid(1, bid);
-    bidResponse.bidderCode = bid.bidder;
-    bidResponse.ad = ad;
-    bidResponse.cpm = cpm;
-    bidResponse.width = bidData.w;
-    bidResponse.height = bidData.h;
-    bidResponse.creativeId = bidData.crid;
-    bidResponse.pubapiId = response.id;
-    bidResponse.currencyCode = response.cur;
-    if (bidData.dealid) {
-      bidResponse.dealId = bidData.dealid;
+      ad += '<script>if(!parent.$$PREBID_GLOBAL$$.aolGlobals.pixelsDropped){' +
+        'parent.$$PREBID_GLOBAL$$.aolGlobals.pixelsDropped=true;' + formattedPixels +
+        '}</script>';
     }
-
-    bidmanager.addBidResponse(bid.placementCode, bidResponse);
   }
 
-  function _isMarketplaceBidder(bidder) {
-    return bidder === AOL_BIDDERS_CODES.aol || bidder === AOL_BIDDERS_CODES.onedisplay;
-  }
+  return {
+    bidderCode: bid.bidderCode,
```

Comment:
Some bidResponse params are missing and invalid. Please check http://prebid.org/dev-docs/bidder-adapter-1.html